src/libraries/System.Private.CoreLib/src/System/Threading/LowLevelMonitor.cs

```diff
@@ -0,0 +1,106 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Diagnostics;
+using System.Runtime.InteropServices;
+
+namespace System.Threading
+{
+    /// <summary>
+    /// Wraps a non-recursive mutex and condition.
+    ///
+    /// Used by the other threading subsystems, so this type cannot have any dependencies on them.
+    /// </summary>
+    [StructLayout(LayoutKind.Auto)]
+    internal partial struct LowLevelMonitor
```

Comment:
I have converted it to struct to avoid unnecessary allocation and indirection on use.